Initial BSP layer definitions

jpetrina opened the pull request:
Pushing initial layer work, with `meta-switchbsp` as platform name, for review.

Thanks!

cc @damirsamardzic

Commit:
initial layer structure

Signed-off-by: Jakov Petrina <jakov.petrina@sartura.hr>

Commit:
README.md: add testing instructions

Signed-off-by: Jakov Petrina <jakov.petrina@sartura.hr>

sonoble commented on conf/machine/arm64-all.conf:
```diff
@@ -0,0 +1,24 @@
+KERNEL_DEVICETREE = " \
+    freescale/fsl-ls1043a-rdb.dtb \ 
+    freescale/fsl-ls1046a-rdb-sdk.dtb \
+    freescale/fsl-ls2080a-rdb.dtb \
```
Is there a developer/customer need for these NXP RDBs? For simplicity they can probably be removed along with any patches unless there is a specific RDB somewhere that is being used for testing.  I believe only the ls1043a-rdb has been used previously and specifically for initial porting work.

damirsamardzic replied:
If these are not needed, we can remove them. Please confirm.

jpetrina replied:
@sonoble, I went ahead and removed these, we can bring them back if they are needed, thanks.

KanjiMonster commented on conf/layer.conf:
```diff
@@ -0,0 +1,22 @@
+    networking-layer:${LAYERDIR}/dynamic-layers/networking-layer/*/*/*.bbappend \
+    meta-linux-mainline:${LAYERDIR}/dynamic-layers/meta-linux-mainline/*/*/*.bb \
+    meta-linux-mainline:${LAYERDIR}/dynamic-layers/meta-linux-mainline/*/*/*.bbappend \
+"
```
These paths (`${LAYERDIR}/dynamic-layers/...`) do not exist, so the `BBFILES_DYNAMIC` definition does not make any sense.

jpetrina replied:
@KanjiMonster, I removed this leftover boilerplate code, thanks!

Commit:
conf: layer: remove unused dynamic layers definitions

Signed-off-by: Jakov Petrina <jakov.petrina@sartura.hr>

Commit:
conf: machine: arm64-all: remove unused NXP RDBs

Signed-off-by: Jakov Petrina <jakov.petrina@sartura.hr>

Commit:
README: fix link identation level

Signed-off-by: Jakov Petrina <jakov.petrina@sartura.hr>

KanjiMonster commented on conf/layer.conf:
```diff
@@ -0,0 +1,13 @@
+BBFILE_PATTERN_switchbsp := "^${LAYERDIR}/"
+BBFILE_PRIORITY_switchbsp = "5"
+
+LAYERDEPENDS_switchbsp = "core"
```
Your README.md says this layer depends meta-oe, but your LAYERDEPENDS does not:

```suggestion
LAYERDEPENDS_switchbsp = "core"
LAYERDEPENDS_switchbsp += "openembedded-layer"
```

jpetrina replied:
Thanks, added. I was surprised to see other 3rd party layers not enforcing this dep.

Commit:
conf: layer: add missing meta-oe layer as dependency

The README mentions this layer as a dependency so enforce this in the
build system.

Reported-by: Jonas Gorski <jonas.gorski@bisdn.de>
Signed-off-by: Jakov Petrina <jakov.petrina@sartura.hr>
